pkg/scanner/library/scan.go

```diff
@@ -56,12 +58,27 @@ func NewScanner(filename string) Scanner {
 	return scanner
 }
 
-func Scan(files extractor.FileMap) (map[string][]vulnerability.DetectedVulnerability, error) {
+func Scan(files extractor.FileMap, scanOptions types.ScanOptions) (map[string][]vulnerability.DetectedVulnerability, error) {
 	results, err := analyzer.GetLibraries(files)
 	if err != nil {
 		return nil, xerrors.Errorf("failed to analyze libraries: %w", err)
 	}
 
+	if !types.IsVulnTypeSelected(scanOptions, "all") && !types.IsVulnTypeSelected(scanOptions, "package") {
+		for path, _ := range results {
+			toDelete := true
+			for i := 0; i < len(scanOptions.VulnType); i++ {
+				depFileDirName := strings.Join([]string{scanOptions.VulnType[i], "app"}, "-")
```

Comment:
This doesn't make much sense. It is not decided where each lock file is located.
e.g. `/myapp/package-lock.json`

`xxx-app` is the specific case in my example.
https://github.com/knqyf263/trivy-ci-test/blob/master/Dockerfile

Reply:
I have deleted this block